hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/DispatchingEvmFrameState.java

```diff
@@ -87,72 +87,90 @@ public class DispatchingEvmFrameState implements EvmFrameState {
     private static final String TOKEN_CALL_REDIRECT_CONTRACT_BINARY =
```

Comment:
It would make me happy to see a comment saying where this magical string came from.  Even better, if the solidity source were checked in somewhere (is it?).

Separately, I wonder if it occurred to anyone that the replacement placeholder need not actually be `fefe...` or any other hex characters, or even the _correct length_ but something that would stand out.  Like `$$TOKEN_ADDRESS$$` or whatever it is.  In fact `%s` would stand out and turn this into a formatting string.  Oh well.

Reply:
Luke showed me that this redirect contract comes from [HIP-719](https://hips.hedera.com/hip/hip-719) - perhaps a comment pointing there, please?

Reply:
Well it actually precedes HIP 719 (probably HIP 376 in 0.26) but HIP 719 is the first place it's laid out properly.
It essentially fuels the ability to treat an HTS token like an ERC token.
So in contract you can omit the tokenAddress as a parameter because you call the contract that "lives" at the token address.
That redirect contract binary picks it up and essentially sets the address as the 1st param in whatever call is being made
e.g. `<htsTokenAddres>.allowance(owner, spender)` =(redirect)=> `IHederaTokenService.allowance(htsTokenAddress, owner, spender)`